llvm/lib/Target/RISCV/RISCVInstrInfo.cpp

```diff
@@ -1907,6 +1907,76 @@ bool RISCVInstrInfo::verifyInstruction(const MachineInstr &MI,
   return true;
 }
 
+bool RISCVInstrInfo::canFoldIntoAddrMode(const MachineInstr &MemI, Register Reg,
+                                         const MachineInstr &AddrI,
+                                         ExtAddrMode &AM) const {
+  switch (MemI.getOpcode()) {
+  default:
+    return false;
+  case RISCV::LB:
+  case RISCV::LBU:
+  case RISCV::LH:
+  case RISCV::LHU:
+  case RISCV::LW:
+  case RISCV::LWU:
+  case RISCV::LD:
+  case RISCV::FLH:
+  case RISCV::FLW:
+  case RISCV::FLD:
+  case RISCV::SB:
+  case RISCV::SH:
+  case RISCV::SW:
+  case RISCV::SD:
+  case RISCV::FSH:
+  case RISCV::FSW:
+  case RISCV::FSD:
+    break;
+  }
+
+  // Check the fold operand is not the loaded/stored value.
+  const MachineOperand &BaseRegOp = MemI.getOperand(0);
```

Comment:
should it be `getOperand(1)`?

Reply:
The name of the variable is confusing but I think the operand number is correct. Operand 0 is the destination for load and the value to store for stores. We're exiting if Reg is equal. This should never be true for a load. So we're only concerned with store. If they are equal for store, it means that Reg isn't the pointer.

Reply:
Oh, ok. I read the context from `MachineSink.cpp`  - the `Reg` is known to be a use in `MemI` and we're still in SSA form. Could we rename the variable to something else? How about `ValueOp`?

Reply:
I don't think we need the isReg check since the value operand should always be a register. So I got rid of the variable.